core/communications/httpCommunication.go

```diff
@@ -1158,36 +1187,27 @@ func (communication *HTTP) handleGetData(orgID string, objectType string, object
 		} else {
 			writer.Header().Add("Content-Type", "application/octet-stream")
 			writer.WriteHeader(http.StatusOK)
+
+			// Start the download
 			if _, err := io.Copy(writer, dataReader); err != nil {
 				SendErrorResponse(writer, err, "", 0)
 			}
 			if err := Store.CloseDataReader(dataReader); err != nil {
 				SendErrorResponse(writer, err, "", 0)
 			}
+
 			if trace.IsLogging(logger.DEBUG) {
-				trace.Debug("Handling object get data, update notification for %s %s %s %s, status: %s\n", objectType, objectID, destType, destID, common.Data)
-			}
-			// update notification only if current notification.InstanceID == metadata.InstanceID && current notification.status == "updated"
-			if updateNotificationRecord {
-				if trace.IsLogging(logger.DEBUG) {
-					trace.Debug("Handling object get data, update notification status to data for %s %s %s %s %s\n", orgID, objectType, objectID, destType, destID)
-				}
-				notification := common.Notification{ObjectID: objectID, ObjectType: objectType,
-					DestOrgID: orgID, DestID: destID, DestType: destType, Status: common.Data, InstanceID: instanceID, DataID: dataID}
-				if err = Store.UpdateNotificationRecord(notification); err != nil {
-					if log.IsLogging(logger.ERROR) {
-						log.Error("Handling object get data, failed to update notification for %s %s %s %s with status: %s\n", objectType, objectID, destType, destID, common.Data)
-					}
-				} else {
-					if trace.IsLogging(logger.DEBUG) {
-						log.Debug("Handling object get data, update notification for %s %s %s %s with status %s is done\n", objectType, objectID, destType, destID, common.Data)
-					}
-				}
-			} else {
-				if trace.IsLogging(logger.DEBUG) {
-					trace.Debug("Handling object get data, return without update notification status to data for %s %s %s %s %s, set updateNotificationRecord to %t \n", orgID, objectType, objectID, destType, destID, updateNotificationRecord)
-				}
+				handleGetData_download_complete_time := time.Now().Unix()
+				trace.Debug("handleGetData has taken %d seconds to downloads %s %s %s %s", (handleGetData_download_complete_time - handleGetData_enter_time), orgID, objectType, objectID, destID)
 			}
+
+			/**
```

Comment:
This code was integral to the last set of performance changes, to ensure that all the models were correctly delivered. Removal of this doesnt open up the possibility of leaving the node in an error state when the model was actually delivered correctly?

Reply:
In my testing that ran, I did not encounter any problems when I removed this code and going up to the 2500 agents again. And I did experience intermittent failures when I had it in.

Reply:
ok